core/types/src/macros.rs

```diff
@@ -42,6 +42,23 @@ macro_rules! new_type_accessors_impls {
     )*}
 }
 
+/// Newtype wrapper for a primitive or struct type
+#[macro_export]
+macro_rules! impl_newtype {
+    ($($wrapper:ident, $inner:ty;)*) => {$(
+        $crate::newtype_accessors_impls! {
+            $wrapper, $inner;
+        }
+
+        impl core::fmt::Display for$wrapper {
+            fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+                core::fmt::Debug::fmt(&self.0, f)
```

Comment:
I didn't like redirecting to `Debug` here but it felt like we would normally want the struct like appearances and it felt unnecessary to implement display for each C type.
Thinking IFF we need we can implement Display on a case by case bases for special C types.